nio/schemas.py

```diff
@@ -17,10 +17,12 @@
 
 from __future__ import unicode_literals
 
+import re
+
 from jsonschema import Draft4Validator, FormatChecker, validators
 
 RoomRegex = "^!.+:.+$"
-UserIdRegex = "^@.*:.+$"
+UserIdRegex = "^@[a-z0-9._=/]+:.+$"
```

Comment:
This regex doesn't cover the characters allowed in [historical user IDs](https://matrix.org/docs/spec/appendices#historical-user-ids), you should use something like `[\x21-\x39\x3B-\x7E]+` for the username part instead. 
A simple way to tighten the server part also, would be to use `[a-zA-Z\d.:-]*[a-zA-Z\d]` instead of `.+`.